src/partition/src/splitter.rs

```diff
@@ -42,12 +43,35 @@ impl WriteSplitter {
         }
     }
 
-    pub fn split_insert(&self, insert: InsertRequest) -> Result<InsertRequestSplit> {
+    pub fn split_insert(
+        &self,
+        insert: InsertRequest,
+        schema: &Schema,
+    ) -> Result<InsertRequestSplit> {
         check_req(&insert)?;
-
-        let column_names = self.partition_rule.partition_columns();
-        let values = &insert.columns_values;
-        let partition_columns = find_partitioning_values(values, &column_names)?;
+        let mut insert = insert;
+        let partition_columns = self.partition_rule.partition_columns();
+        for column_schema in schema.column_schemas() {
+            if partition_columns.contains(&column_schema.name)
+                && (!insert.columns_values.contains_key(&column_schema.name))
+            {
+                let key = insert.columns_values.keys().next().unwrap();
+                let row_nums = insert.columns_values.get(key).unwrap().len();
```

Comment:
The row numbers are checked in method `check_req` above. Can you return the row number from `check_req` and use it here?